sensor/common/detector/queue/queue.go

```diff
@@ -61,16 +61,12 @@ func (q *Queue[T]) Push(item T) {
 
 func (q *Queue[T]) run() {
 	defer close(q.outputC)
-	for {
-		select {
-		case <-q.stopper.Flow().StopRequested():
-			return
-		case <-q.isRunning.Done():
-			select {
-			case <-q.stopper.Flow().StopRequested():
-				return
-			case q.outputC <- q.queue.PullBlocking(q.stopper.LowLevel().GetStopRequestSignal()):
-			}
+	select {
+	case <-q.stopper.Flow().StopRequested():
+		return
+	case <-q.isRunning.Done():
```

Comment:
can we remove this select and combine the contexts

Reply:
Looks like no, we wait for isRunning but there is an option to stop waiting on flow request